datadayessentials/data_transformation/_transformers.py

```diff
@@ -386,89 +386,69 @@ def missing_column_warning(self, columns_to_evaluate):
-        for col in self.categorical_columns_to_split:
-            if "QCB" in col:
-                cat_series, numerical_series = self._split_categorical_column(
-                    df_in[col]
-                )
-                num_dict[col + "_num"] = numerical_series
-                df_in[col] = cat_series
-        df_in = pd.concat([df_in, pd.DataFrame(num_dict)], axis=1)
-        return df_in
-
-
-class InferenceSpeedCategoricalColumnSplitter:
+class CategoricalColumnSplitter(IDataFrameTransformer):
+    numerical_mapping = {"D": 5, "R": 6, "V": 6, "S": 0, "A": 2}
+    cat_mapping = {0: np.nan, 1: np.nan, 2: np.nan, 3: "D", 4: "D", 5: "D", 6: "D", '0': np.nan, '1': np.nan, '2': np.nan, '3': "D", '4': "D", '5': "D", '6': "D"}
     def __init__(self, categorical_columns_to_split: list):
         self.categorical_columns_to_split = categorical_columns_to_split
 
-    @staticmethod
-    def _inference_split_categorical_column(series: pd.Series, force_numeric: bool = True) -> tuple[pd.Series, pd.Series]:
-        # Mapping for numerical and categorical transformations
-        numerical_mapping = {"D": 5, "R": 6, "V": 6, "S": 0, "A": 2}
-        cat_mapping = {0: np.nan, 1: np.nan, 2: np.nan, 3: "D", 4: "D", 5: "D", 6: "D"}
+    def process(self, data: Union[pd.DataFrame, pd.Series]):
+        if isinstance(data, pd.DataFrame):
+            return self.process_df(data)
+        elif isinstance(data, pd.Series):
+            return self.process_series(data)
+        else:
+            raise TypeError(f"Expected DataFrame or Series, got {type(data)}")
 
+    def _inference_split_categorical_column(self, series: pd.Series) -> tuple[pd.Series, pd.Series]:
         # Apply transformations
-        numerical_series = series.map(numerical_mapping).fillna(series)
-        cat_series = series.map(cat_mapping).fillna(series)
+        numerical_series = series.map(self.numerical_mapping).fillna(series).astype('float64')
 
-        # Convert the numerical series to numeric if required
-        if force_numeric:
-            numerical_series = pd.to_numeric(numerical_series, errors='coerce')
+        cat_series = series.copy(deep=True).replace(self.cat_mapping)
+        cat_mask = pd.to_numeric(cat_series, errors='coerce').isna()
+        cat_series[~cat_mask] = np.nan
+        cat_series = cat_series.replace(self.cat_mapping).astype('object')
+    
 
         return cat_series, numerical_series
 
-    def process(self, df_in: pd.DataFrame) -> pd.DataFrame:
+    def process_df(self, df_in: pd.DataFrame) -> pd.DataFrame:
         for column in self.categorical_columns_to_split:
-            if 'QCB' in column and column in df_in.columns:
+            if ('QCB' in column) and (column in df_in.columns):
                 cat_series, numerical_series = self._inference_split_categorical_column(df_in[column])
 
                 # Update DataFrame in place
                 df_in[f"{column}"] = cat_series
                 df_in[f"{column}_num"] = numerical_series
```

Comment:
we should be passing the correct columns there and not relying on the column validation.

Reply:
Ah thats what you meant! Yes will take it away